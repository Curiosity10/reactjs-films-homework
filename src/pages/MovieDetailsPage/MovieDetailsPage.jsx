import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Header from '../../components/Header';
import HeroContainer from '../../components/Hero';
import Menu from '../../components/Menu';
import MoviesContainer from '../../components/Movies';
import styles from './MovieDetailsPage.scss';
import Footer from '../../components/Footer';

const MovieDetailsPage = ({
  fetchMovieDetails,
  fetchMoviesByFilter,
  fetchGenres,
  changeFilter,
  changeLayout,
  id,
}) => {
  const [isModalOpen, setModalOpen] = useState(false);

  useEffect(() => {
    changeLayout();
    changeFilter();
    fetchMoviesByFilter();
    fetchGenres();
  }, [fetchMoviesByFilter, fetchGenres, changeFilter, changeLayout]);

  useEffect(() => {
    fetchMovieDetails(id);
  }, [id, fetchMovieDetails]);

  return (

    <div className={styles.container}>
      <Header />
      <div className={styles.backdrop}>
        <HeroContainer toggleModal={setModalOpen} />
        <Menu />
        <MoviesContainer isModalOpen={isModalOpen} toggleModal={setModalOpen} />
      </div>
      <Footer />
    </div>
  );
};

MovieDetailsPage.propTypes = {
  fetchMovieDetails: PropTypes.func.isRequired,
  fetchMoviesByFilter: PropTypes.func.isRequired,
  fetchGenres: PropTypes.func.isRequired,
  changeFilter: PropTypes.func.isRequired,
  changeLayout: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
};

export default MovieDetailsPage;
