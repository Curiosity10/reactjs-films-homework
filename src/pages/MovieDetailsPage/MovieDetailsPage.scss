.backdrop {
  background-color: #f5f6f6;
  flex: 1;
}

.container {
  display: flex;
  flex-direction: column;
  min-height: 100%;
}
