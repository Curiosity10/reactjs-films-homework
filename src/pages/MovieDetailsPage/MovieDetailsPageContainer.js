import { connect } from 'react-redux';
import MovieDetailsPage from './MovieDetailsPage';
import {
  fetchMovieDetails,
  fetchMoviesByFilter,
  fetchGenres,
  changeFilter,
} from '../../modules/app/appActions';
import uiActions from '../../modules/ui/uiActions';

export default connect(null, {
  fetchMovieDetails,
  fetchMoviesByFilter,
  fetchGenres,
  changeFilter,
  changeLayout: uiActions.changeLayout,
})(MovieDetailsPage);
