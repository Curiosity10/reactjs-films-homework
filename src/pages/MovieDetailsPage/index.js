import MovieDetailsPage from './MovieDetailsPage';
import MovieDetailsPageContainer from './MovieDetailsPageContainer';

export { MovieDetailsPage, MovieDetailsPageContainer };
