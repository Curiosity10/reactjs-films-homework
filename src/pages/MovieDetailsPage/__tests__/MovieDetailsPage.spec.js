import React from 'react';
import { create, act } from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import MovieDetailsPage from '../MovieDetailsPage';

describe('MovieDetailsPage renders correctly', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    app: {
      movies: [],
      movie: {
        title: 'Test',
        backdrop_path: '/123',
        id: '123',
      },
      genres: [],
      isLoading: true,
      currentPage: 1,
      error: false,
      filter: 'Trending',
      genreId: null,
      searchQuery: '',
      videoKey: '',
      pagesCount: 1,
    },
    ui: {
      layout: 'grid',
    },
  };
  const store = mockStore(initialState);

  it('MovieDetailsPage render correctly', () => {
    let tree;
    const mockChangeFilter = jest.fn();
    const mockChangeLayout = jest.fn();
    const mockFetchGenres = jest.fn();
    const mockFetchMovieDetails = jest.fn();
    const mockFetchMovies = jest.fn();

    act(() => {
      tree = create(
        <StaticRouter location="someLocation">
          <Provider store={store}>
            <MovieDetailsPage
              fetchMovieDetails={mockFetchMovieDetails}
              fetchMoviesByFilter={mockFetchMovies}
              fetchGenres={mockFetchGenres}
              changeFilter={mockChangeFilter}
              changeLayout={mockChangeLayout}
              id="123"
            />
          </Provider>
        </StaticRouter>,
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
    expect(mockFetchMovieDetails).toHaveBeenCalled();
    expect(mockFetchMovies).toHaveBeenCalled();
    expect(mockFetchGenres).toHaveBeenCalled();
    expect(mockChangeFilter).toHaveBeenCalled();
    expect(mockChangeLayout).toHaveBeenCalled();
  });
});
