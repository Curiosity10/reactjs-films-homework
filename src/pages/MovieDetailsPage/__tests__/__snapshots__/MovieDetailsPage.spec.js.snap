// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsPage renders correctly MovieDetailsPage render correctly 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <h1>
        <a
          className="logo"
          href="/"
          onClick={[Function]}
        >
          Films
        </a>
      </h1>
      <form
        className="searchForm"
        onSubmit={[Function]}
      >
        <input
          aria-label="Search Input"
          className="input"
          onChange={[Function]}
          placeholder="Search..."
          type="search"
          value=""
        />
        <button
          aria-label="Search Button"
          className="searchBtn"
          type="submit"
        />
      </form>
    </div>
  </header>
  <div
    className="backdrop"
  >
    <div
      className="container"
    >
      <div
        className="spinner"
      >
        <div
          className="ellipsis"
        >
          <div />
          <div />
          <div />
          <div />
        </div>
        <p
          className="title"
        >
          Loading
        </p>
      </div>
    </div>
    <section
      className="menu"
    >
      <nav>
        <ul>
          <li
            className="navItem"
          >
            <button
              aria-label="Trending"
              className="btn"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Trending
            </button>
          </li>
          <li
            className="navItem"
          >
            <button
              aria-label="Top Rated"
              className="btn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Top Rated
            </button>
          </li>
          <li
            className="navItem"
          >
            <button
              aria-label="Coming Soon"
              className="btn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Coming Soon
            </button>
          </li>
          <li
            className="navItem"
          >
            <select
              aria-label="Genres"
              className="selectGenre"
              name="selectGenre"
              onChange={[Function]}
              value=""
            >
              <option
                disabled={true}
                hidden={true}
                value=""
              >
                Genre
              </option>
            </select>
          </li>
        </ul>
      </nav>
      <div>
        <button
          aria-label="Grid"
          className="btn btn__active"
          onClick={[Function]}
          type="button"
        />
        <button
          aria-label="Table"
          className="btn"
          onClick={[Function]}
          type="button"
        />
      </div>
    </section>
    <main
      className="container"
    />
    <div
      className="spinner"
    >
      <div
        className="ellipsis"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <p
        className="title"
      >
        Loading
      </p>
    </div>
  </div>
  <footer
    className="footer"
  >
    <h2
      className="logo"
    >
      Films
    </h2>
    <p
      className="copyright"
    >
      Copyright Â© 2019 FILMS. DZMITRY
    </p>
  </footer>
</div>
`;
